src/WasmDis.ts

```diff
@@ -1171,3 +1237,185 @@ export class NameSectionReader {
     return new NameSectionNameResolver(functionNames, this._functionLocalNames);
   }
 }
+
+export class DevToolsNameGenerator {
+  private _done: boolean;
+  private _hasNames: boolean;
+  private _functionImportsCount: number;
+  private _memoryImportsCount: number;
+  private _tableImportsCount: number;
+  private _globalImportsCount: number;
+
+  private _functionNames: string[][];
+  private _functionLocalNames: string[][];
+  private _memoryNames: string[][];
+  private _tableNames: string[][];
+  private _globalNames: string[][];
+
+  constructor() {
+    this._done = false;
+    this._hasNames = false;
+    this._functionImportsCount = 0;
+    this._memoryImportsCount = 0;
+    this._tableImportsCount = 0;
+    this._globalImportsCount = 0;
+
+    this._functionNames = null;
+    this._functionLocalNames = null;
+    this._memoryNames = null;
+    this._tableNames = null;
+    this._globalNames = null;
+  }
+
+  private _generateExportName(field: Uint8Array) : string {
+    const exportName = bytesToString(field).replace(/\s/g, '_');
+    return exportName.replace(/\s/g, '_');
+  }
+
+  private _generateImportName(moduleName: Uint8Array, field: Uint8Array) : string {
+    if (moduleName.length == 0 && field.length == 0)
+      return null;
+    const name = bytesToString(moduleName) + '.' + bytesToString(field);
+    return name.replace(/\s/g, '_');
+  }
+
+  private _setName(names : string[][], index : number, nameSource: NameSource, name : string) {
+    if (!names[index])
+      names[index] = [];
+
+    // If the same entity is used in several exports, use the first name encountered.
+    if (NameSource.Export == nameSource) {
+      if (names[index][NameSource.Export]) return;
+    }
+
+    names[index][nameSource] = name;
+  }
+
+  public read(reader: BinaryReader): boolean {
+    if (this._done)
+    throw new Error('Invalid state: disassembly process was already finished.')
+    while (true) {
+      if (!reader.read())
+        return false;
+      switch (reader.state) {
+        case BinaryReaderState.END_WASM:
+          if (!reader.hasMoreBytes()) {
+            this._done = true;
+            return true;
+          }
+          break;
+        case BinaryReaderState.ERROR:
+          throw reader.error;
+        case BinaryReaderState.BEGIN_WASM:
+          this._functionImportsCount = 0;
+          this._memoryImportsCount = 0;
+          this._tableImportsCount = 0;
+          this._globalImportsCount = 0;
+
+          this._functionNames = [];
+          this._functionLocalNames = [];
+          this._memoryNames = [];
+          this._tableNames = [];
+          this._globalNames = [];
+          break;
+        case BinaryReaderState.END_SECTION:
+          break;
+        case BinaryReaderState.BEGIN_SECTION:
+          var sectionInfo = <ISectionInformation>reader.result;
+
+          if (sectionInfo.id === SectionCode.Custom &&
+              bytesToString(sectionInfo.name) === "name") {
```

Comment:
Is "name" the default, and is there a better way to check for it?

Reply:
According to [https://webassembly.github.io/spec/core/appendix/custom.html](this) the name section should be called "name". Currently this is also the way it is done in the NameResolver class (so I copied it) and looking up in wabt, they also seem to check whether the section is called "name". I'm not aware of another way of detecting this. As far as I can see from the referenced link above the only way to distinguish the name section from another section is the combination of section id + the "name" name.

Reply:
I see `"name"` used at least two times. Can we create a constant for that (perhaps `NAME_SECTION_ID`)?

Reply:
Sure! Will upload a patch.

Reply:
Thanks for the review! I have added a NAME_SECTION_NAME field in my latest patch.  "NAME_SECTION_ID" may be confusing since there's already a section id. If you don't agree, I can change it :).